src/menus/picklist/picklist.ts

```diff
@@ -37,11 +38,35 @@ export class NglPicklist {
   @Input() open: boolean;
   @Output() openChange = new EventEmitter();
 
+  @Input() filter: string | Function;
+  @Input() filterPlaceholder = 'Filter Options';
+
+  get hasFilter() {
+    return typeof(this.filter) !== 'undefined';
+  }
+
+  get filteredData() {
+    const filter = <any>this.filter; // until proper type guards are available in typescript
+    switch (typeof(this.filter)) {
+      case 'string':
+        if (/^\s*$/.test(filter)) {
+          return (this.data || []).filter(d => (d || {}).toString().toLowerCase().indexOf(this._filter) !== -1);
+        }
+        return (this.data || []).filter(d => (d[filter] || '').toLowerCase().indexOf(this._filter) !== -1);
+      case 'function':
+        return (this.data || []).filter(filter);
+      default:
+        return this.data;
```

Comment:
Do you think it would be better to notify user with an error/warning that we don't support another option?